Release/src/uri/uri.cpp

```diff
@@ -19,6 +19,9 @@ namespace web { namespace details
 {
 namespace
 {
+    const ::utility::string_t dotSegment = _XPLATSTR(".");
+    const ::utility::string_t dotDotSegment = _XPLATSTR("..");
```

Comment:
I would argue that in this case global static constants do more harm than good.
While either way they should not allocate memory thanks to small string optimization, compiler generates atexit handlers for each global object which bloats the binary. See [this example](https://godbolt.org/z/VanYIu): notice for each object `dot` and `dotDot` an atexit handler `??__Fdot@@YAXXZ` and `??__FdotDot@@YAXXZ` is generated, and that's besides static initialization code.
Ultimately it's a trade off, but I would prefer to construct two strings (assuming small string optimization) every once in a while, than to have extra code that works every time, takes up space in a binary, but in the end may not be used every time.

Reply:
Fair enough, that change is reverted in https://github.com/Microsoft/cpprestsdk/commit/5a9f3bc25714dfbfea7fd006085a1f18ee6acde2